torchacc/dist/distributed_parallel.py

```diff
@@ -35,32 +34,38 @@ def __init__(self, model: torch.nn.Module, config: Config, **kwargs):
             need_wrap_dp = self.has_dp and not self.has_tp
 
         if need_wrap_dp:
-            if self.model is None:
-                self.model = DataParallel(model, self.config, **kwargs)
+            if self._module is None:
+                self._module = DataParallel(model, self.config, **kwargs)
             else:
-                model = self.model._get_underlay_model()
-                model = DataParallel(model, self.config, **kwargs)
-                self.model._update_underlay_model(model)
+                module = self._module._get_underlay_model()
+                module = DataParallel(model, self.config, **kwargs)
+                self._module._update_underlay_model(module)
+
+        if self._module is None:
+            self._module = module
 
-        if self.model is None:
-            self.model = model
+    def __getattr__(self, name):
+        try:
+            return super().__getattr__(name)
+        except AttributeError:
+            return self._get_underlay_model().__getattr__(name)
```

Comment:
better use `getattr(self._get_underlay_model(), name)` here, some class do not implement `__getattr__`